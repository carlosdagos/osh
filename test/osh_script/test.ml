(* This module tests that we can create inline scripts
   which are of type Osh.Proc.t *)

[%%osh_def ["cat"; "awk"; "grep"; "echo"]]


let%osh_script my_script =
      echo ["Hello World!"]
  ||> grep ["hello"; "-i"]
  ||> awk  ["{print $1}"]


let%osh_script hello_world_script ?(message="hello_world") () =
  let message = message ^ " appended" in
  let echo = echo [message] in
  echo

let%expect_test _ =
  let open Osh.Proc in
  let message = "Hello Osh!" in

  run_process' my_script;
  run_process' (hello_world_script ~message ());
  [%expect{|
  Hello
  Hello Osh! appended
  |}]
