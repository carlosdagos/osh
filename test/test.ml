let () =
  let open Osh.Proc in
  run_process' begin
        [%osh "echo", ["hello world"]]
    ||> [%osh "wc", ["-l"]]
  end
