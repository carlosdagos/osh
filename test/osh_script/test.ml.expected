let cat args = Osh.Proc.create_sys_process ~program:"cat" ~args
let awk args = Osh.Proc.create_sys_process ~program:"awk" ~args
let grep args = Osh.Proc.create_sys_process ~program:"grep" ~args
let echo args = Osh.Proc.create_sys_process ~program:"echo" ~args
let my_script =
  let open Osh.Proc in
    Osh.Proc.identity
      (((echo ["Hello World!"]) ||> (grep ["hello"; "-i"])) ||>
         (awk ["{print $1}"]))
let hello_world_script =
  let open Osh.Proc in
    fun ?(message= "hello_world") ->
      fun () ->
        Osh.Proc.identity
          (let message = message ^ " appended" in
           let echo = echo [message] in echo)
[%%expect_test
  let _ =
    let open Osh.Proc in
      let message = "Hello Osh!" in
      run_process' my_script;
      run_process' (hello_world_script ~message ());
      [%expect {|
  Hello
  Hello Osh! appended
  |}]]
